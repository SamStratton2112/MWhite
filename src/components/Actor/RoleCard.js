import { Col, Container} from 'react-bootstrap';
import {useState} from 'react';

const RoleCard = ({role,name,company,year,location}) => {
    const [hidden, setHidden] = useState(true);
    const body1 = <h4 style={{ paddingTop:'45%' }}>{name}</h4>;
    const body2 = (
        <Container style={{textAlign:'left'}}>
        <h4 style={{marginTop:'1em'}}>{name}</h4>
        <p><b>Role : </b>{role}</p>
        <p><b>Company : </b>{company}</p>
        <p><b>Year : </b>{year}</p>
        <p><b>Location : </b>{location}</p>
        </Container>
    );
    return (
        <Col>
            <Container 
                style={{ width: '15em', height: '17em', border: 'solid', margin: '1em', borderRadius:'.3em', overflowY: 'auto', textAlign:'center'}} 
                className="cardBg" 
                onMouseEnter={() => setHidden(false)} 
                onMouseLeave={() => setHidden(true)}
            >
                {hidden ? body1 : body2}
            </Container>
        </Col>
    );
};

export default RoleCard;