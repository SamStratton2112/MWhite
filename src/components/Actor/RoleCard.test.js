import RoleCard from './RoleCard';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

describe(RoleCard, () => {
    it("renders without crashing", () => {
        render(<MemoryRouter>
            <RoleCard />
        </MemoryRouter>);
    });
    it("matches snapshot", () => {
        const { asFragment } = render(<MemoryRouter>
            <RoleCard />
        </MemoryRouter>);
        expect(asFragment()).toMatchSnapshot();
    });
});