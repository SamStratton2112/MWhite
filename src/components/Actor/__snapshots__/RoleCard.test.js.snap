// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoleCard matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="col"
  >
    <div
      class="cardBg leftText container"
      style="width: 17em; height: 17em; border: solid; margin: 1em; border-radius: .3em; overflow-y: auto; text-align: center;"
    >
      <h4
        style="padding-top: 45%;"
      />
    </div>
  </div>
</DocumentFragment>
`;
